src/fast_forward.rs

```diff
@@ -109,7 +109,7 @@ pub fn fast_forward_singleton(
 
     // this is the tree hash of the singleton top layer puzzle
     // the tree hash of singleton_top_layer_v1_1.clsp
-    if singleton.args.singleton_struct.mod_hash != SINGLETON_TOP_LAYER_PUZZLE_HASH {
+    if singleton.args.singleton_struct.mod_hash.into_inner() != SINGLETON_TOP_LAYER_PUZZLE_HASH {
```

Comment:
this is moving the inner `Vec<u8>` out of `singleton.args.singleton_struct.mod_hash`, right?
maybe it doesn't make a difference, but I would have expected something like `.as_ref()`, to have a non-destructive comparison

Reply:
Same as above, since it implements `Copy`, `into_inner` isn't actually dropping the original value. This might be a confusing name for the function though, since typically `into_inner` is reserved for `Owned => Owned` conversions.

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Maybe `to_array` would be more appropriate?

Reply:
In this case I don't know that the copying of 32 bytes will matter all that much, but I can change it to `as_ref()` just in case.